core-relations/src/free_join/mod.rs

```diff
@@ -600,20 +633,41 @@ impl Database {
     /// Get direct mutable access to the table.
     ///
     /// This method is useful for out-of-band access to databse state.
-    pub fn get_table(&self, id: TableId) -> &WrappedTable {
+    ///
+    // Not in doc comment since it mentions private items:
+    //
+    // **NOTE:** It is legal to call [`Table::new_buffer`] on the returned table handle, and use
```

Comment:
Why is this "legal" and only "likely" a source of bugs? If a table is updated this way solely, are such updates always forgotten?

Reply:
Thanks for the clarification! It would be great if we could incorporate part of your comments into the documentation (e.g., the assumption that downstream crates shouldn't expose `Database` (and `Table`?) to users).